### PR TITLE
 Modularize nir parser out of tools

### PR DESCRIPTION
**Pull request by densh**
This removes fastparse dependency from tools jar.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -231,19 +231,32 @@ lazy val nir =
...
     .settings(toolSettings)
     .settings(mavenPublishSettings)
     .settings(
       libraryDependencies ++= Seq(
-        "com.lihaoyi"    %% "fastparse"  % "0.4.2",
-        "com.lihaoyi"    %% "scalaparse" % "0.4.2",
         "org.scalacheck" %% "scalacheck" % "1.13.4" % "test",
-        compilerPlugin(
-          "org.scalamacros" % "paradise" % "2.1.0" cross CrossVersion.full),
-        "org.scalatest" %% "scalatest" % "3.0.0" % "test"
```
Comment (Duhemm) [inline build.sbt]: I think that you can also remove this from `project/build.sbt`:

https://github.com/scala-native/scala-native/blob/682dab82c7c689d00ebe6d263685694994287ad4/project/build.sbt#L17-L20

File: build.sbt [hunk 2]
```diff
@@ -231,19 +231,32 @@ lazy val nir =
     .settings(mavenPublishSettings)
     .dependsOn(util)
 
+lazy val nirparser =
```
Comment (Duhemm) [inline build.sbt]: Also, this project should be added to `test-all` and `scala-native/scripts/release`

File: build.sbt (hunk 1 above)
Comment (densh) [reply build.sbt]: Indeed. 👍

File: build.sbt (hunk 2 above)
Comment (densh) [reply build.sbt]: Good catch on including in `test-all`. We don't have to release it yet, as it's not part of the public API at the moment.

Comment (densh) [reply build.sbt]: a6e6257

File: build.sbt (hunk 1 above)
Comment (densh) [reply build.sbt]: a6e6257

**Pull request merged**